src/mutex.rs

```diff
@@ -268,6 +268,9 @@ pub struct Lock<'a, T: ?Sized> {
     acquire_slow: Option<AcquireSlow<&'a Mutex<T>, T>>,
 }
 
+unsafe impl<T: Send + ?Sized> Send for Lock<'_, T> {}
+unsafe impl<T: Sync + ?Sized> Sync for Lock<'_, T> {}
+
```

Comment:
I didn't mean to split the changes based on files. I was talking about this (addition of new implementations) being a separate logical change (than loosening of Send and Sync) and hence asked for the commit to be split. The commit messages will be different because they will summarize 2 different changes.

[This](https://www.aleksandrhovhannisyan.com/blog/atomic-git-commits/) is a very good read if you are new to idea of atomic commits.

---

P.S. Unfortunately Github doesn't notify for you pushing new changes or resolving threads. You'll need to hit the "Re-request review" button next to my name on the top right after addressing review comments.

Reply:
The commit messages accurately describe the changes. For the `Mutex` impls, the manual `Send`/`Sync` impls replace those that the compiler would automatically generate otherwise. So the overall effect is to loosen the bounds on existing `Send`/`Sync` impls, in both cases. To describe them as completely new impls would be incorrect; `Send` and `Sync` impls are already present, whether or not they appear in the source.

Reply:
> The commit messages accurately describe the changes.

If two commits have exactly the same log message, they are exactly the same logical change and hence shouldn't be split.

> For the `Mutex` impls, the manual `Send`/`Sync` impls replace those that the compiler would automatically generate otherwise. So the overall effect is to loosen the bounds on existing `Send`/`Sync` impls, in both cases. To describe them as completely new impls would be incorrect; `Send` and `Sync` impls are already present, whether or not they appear in the source.

Thanks for explaining. So they're the same logical change (i-e the addition of new impls would not be a good change w/o the other changes)? If so, I was wrong and I wish you had pointed this out instead of splitting the commit. The point of reviews is not to satisfy the reviewers' wishes but to address problems, if any. :)

Anyway, if they're the same logical change, please squash them back together.